google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/GoogleCloudApiManagementPanel.java

```diff
@@ -0,0 +1,278 @@
+  private static final int CLOUD_API_NAME_COL = 0;
+  private static final int CLOUD_API_ENABLEMENT_COL = 1;
+
+  GoogleCloudApiManagementPanel() {
+    apisPanel.setBorder(BorderFactory.createEmptyBorder());
+
+    selectedApisWrapper.setBorder(IdeBorderFactory.createBorder());
+    selectedApisPanel.setBorder(
+        IdeBorderFactory.createTitledBorder(
+            GctBundle.message("cloud.apis.management.selectedapis.title")));
+
+    allApisWrapper.setBorder(IdeBorderFactory.createBorder());
+    allApisPanel.setBorder(
+        IdeBorderFactory.createTitledBorder(
+            GctBundle.message("cloud.apis.management.allapis.title")));
+
+    if (projectSelector.getSelectedProject() != null) {
+      showApiEnablementUi();
+    }
+    projectSelector.addProjectSelectionListener(project -> showApiEnablementUi());
+  }
+
+  private void showApiEnablementUi() {
+    apisScrollPane.setVisible(true);
+    selectProjectLabel.setVisible(false);
+  }
+
+  /** Initializes the table models of the API tables based on the user's library selection. */
+  // TODO(eshaul) its very important that we add an equals method to CloudLibrary so that the set
+  // operation is reliable
+  void init(List<CloudLibrary> allLibraries, Set<CloudLibrary> selectedLibraries) {
+    selectedApisTable.setModel(new ApisTableModel(selectedLibraries));
+    selectedApisTable
+        .getColumnModel()
+        .getColumn(CLOUD_API_NAME_COL)
+        .setMinWidth(MIN_CLOUD_API_COL_WIDTH);
+    selectedApisTable
+        .getColumnModel()
+        .getColumn(CLOUD_API_ENABLEMENT_COL)
+        .setMinWidth(MIN_CLOUD_API_ENABLEMENT_COL_WIDTH);
+
+    allApisTable.setModel(
+        new ApisTableModel(Sets.difference(Sets.newHashSet(allLibraries), selectedLibraries)));
+    allApisTable
+        .getColumnModel()
+        .getColumn(CLOUD_API_NAME_COL)
+        .setMinWidth(MIN_CLOUD_API_COL_WIDTH);
+    allApisTable
+        .getColumnModel()
+        .getColumn(CLOUD_API_ENABLEMENT_COL)
+        .setMinWidth(MIN_CLOUD_API_ENABLEMENT_COL_WIDTH);
+  }
+
+  JPanel getPanel() {
+    return panel;
+  }
+
+  @VisibleForTesting
+  JBScrollPane getApisScrollPane() {
+    return apisScrollPane;
+  }
+
+  @VisibleForTesting
+  JLabel getSelectProjectLabel() {
+    return selectProjectLabel;
+  }
+
+  @VisibleForTesting
+  JTable getSelectedApisTable() {
+    return selectedApisTable;
+  }
+
+  @VisibleForTesting
+  JTable getAllApisTable() {
+    return allApisTable;
+  }
+
+  @VisibleForTesting
+  ProjectSelector getProjectSelector() {
+    return projectSelector;
+  }
+
+  private void createUIComponents() {
+    selectedApisTable = new ApisTable();
+    selectedApisTable.setTableHeader(null);
+
+    allApisTable = new ApisTable();
+    allApisTable.setTableHeader(null);
+  }
+
+  private static final class ApisTable extends JBTable {
+    ApisTable() {
+      setDefaultRenderer(CloudLibrary.class, new CloudLibraryRenderer());
```

Comment:
Do we only create separate `ApisTable` class to `setDefaultRenderer`? Maybe leave it as general JBTable and just call the methods directly in  `createUIComponents`?

Reply:
yea good point. that class is a lightweigt

Reply:
actually the main reason is that I am creating two of these api tables which means that with an `ApisTable` implementation I can then set the renderer only once (encapsulated in the table). wdyt?

Reply:
Makes sense then, thanks